junit5/src/main/java/sunstone/api/WithAwsCfTemplate.java

```diff
@@ -15,22 +15,42 @@
 /**
  * Deploy CloudFormation template
  * <p>
- * Deployed as a stack in {@link BeforeAllCallback} and stack is deleted in {@link  AfterAllCallback}
+ * Deployed as a stack in {@link BeforeAllCallback} and whole stack is deleted in {@link  AfterAllCallback} or
+ * once the suite is finished (see {@link WithAwsCfTemplate#perSuite()})
  */
 @Retention(RetentionPolicy.RUNTIME)
 @Target({ElementType.TYPE})
-@Repeatable(WithAwsCfTemplateRepetable.class)
+@Repeatable(WithAwsCfTemplateRepeatable.class)
 @ExtendWith({SunstoneExtension.class})
 @Inherited
 public @interface WithAwsCfTemplate {
     /**
-     * Template file located in resources.
+     * Template file located in resources
      */
     String template();
 
+    /**
+     * Region that should be used for creating resource group. Expression is allowed, e.g. {@code abc-${var:default}-xyz} -
+     * var is resolved from system properties.
+     *
+     * If empty, {@code junit5.az.region} from {@code sunstone.properties} is used.
+     */
+    String region() default "";
+
     /**
      * Array of parameters
      * Values may can be an expression - 'value-${variable:default}' - var is resolved from system properties.
     */
     Parameter[] parameters() default {};
+
+    /**
+     * <p>
+     * True if resources supposed to be managed at suite level.
+     * <p>
+     * The template is deployed only once and undeployed once the suite is done.
+     * <p>
+     * Suite is a set of test classes that runs in a bulk not interfering with other suites (sets of test classes).
+     * Suite may be defined as a surefire run or as {@link org.junit.platform.suite.api.Suite}
+     */
+    boolean perSuite() default false;
```

Comment:
I need to talk about this concept.

- This is pretty new concept. I am not aware we use that before in team. Once we thought about something like this in OpenShift testing, but we did not get there. Because OpenShift setup is still faster than hyperscaler setup? Do hyperscalers have something similar as we have namespaces in OpenShift? Anyway That opens new set of new questions which we do not ask before.
- Will there be any sort of cleaning between tests?
- Can order of tests influence results of tests?
- It does rely on no paralel execution?
- How exactly is this implemented? Is it safe? It is stored somewhere unique template name was performed for given Suite.

Reply:
Already ansered in the call.